lib/Jsrt/JsrtHelper.cpp

```diff
@@ -89,7 +89,7 @@ void JsrtCallbackState::ObjectBeforeCallectCallbackWrapper(JsObjectBeforeCollect
         pthread_key_create(&s_threadLocalDummy, DISPOSE_CHAKRA_CORE_THREAD);
 #endif
 
-#if defined(CHAKRA_STATIC_LIBRARY)
+#if defined(CHAKRA_STATIC_LIBRARY) || !defined(_WIN32)
```

Comment:
@jianchun if you read the code and comments carefully, that is used in order to make sure we clean the things for the thread on shared library (linux) Before that code, it was leaking..

As for the PAL thing. I would like step a bit back. Currently the only reason for enabling this trick is to enable PAL lazily from outside right? Eventually, what I understood from both the messages here and email I've got that you do this to make some unit tests passing temporarily on other project right? However, I'm not sure if introducing this unintended change to main code is a really good approach.

Before merging this, it could be nice to have some details on what is actually failing and how etc.